src/components/PopupWithForm.js

```diff
@@ -7,44 +7,42 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._inputs = this._popupElement.querySelectorAll(".modal__input");
     this._submitButton = this._popupElement.querySelector(
-      ".modal__submit-button"
+      ".modal__button"
     );
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
   }
 
   _getInputValues() {
     const _inputs = Array.from(

```

Comment:
Needs correction: There is no need to start the local value with `_`. We use that prefix only for the private class instance variables and methods.